Src/PerformanceCollector/Unit.Tests/QuickPulse/QuickPulseTelemetryProcessorTests.cs

```diff
@@ -694,5 +710,300 @@ public void QuickPulseTelemetryProcessorDoesNotCollectFullTelemetryItemsWhenSwit
+            Assert.AreEqual(
+                ((DependencyTelemetryDocument)accumulatorManager.CurrentDataAccumulator.TelemetryDocuments.ToList()[1]).Name,
+                dependencyLong.Name.Substring(0, MaxFieldLength));
+        }
+
+        [TestMethod]
+        public void QuickPulseTelemetryProcessorTruncatesLongFullDependencyTelemetryItemProperties()
+        {
+            // ARRANGE
+            var accumulatorManager = new QuickPulseDataAccumulatorManager();
+            var telemetryProcessor = new QuickPulseTelemetryProcessor(new SimpleTelemetryProcessorSpy());
+            var instrumentationKey = "some ikey";
+            ((IQuickPulseTelemetryProcessor)telemetryProcessor).StartCollection(
+                accumulatorManager,
+                new Uri("http://microsoft.com"),
+                new TelemetryConfiguration() { InstrumentationKey = instrumentationKey });
+
+            // ACT
+            var dependencyShort = new DependencyTelemetry(
+                "dependencyShort",
+                "dependencyShort",
+                DateTimeOffset.Now,
+                TimeSpan.FromSeconds(1),
+                false)
+            {
+                Properties = { { new string('p', MaxFieldLength), new string('v', MaxFieldLength) } },
+                Context = { InstrumentationKey = instrumentationKey }
+            };
+
+            var dependencyLong = new DependencyTelemetry(
+                "dependencyLong",
+                "dependencyLong",
+                DateTimeOffset.Now,
+                TimeSpan.FromSeconds(1),
+                false)
+            {
+                Properties = { { new string('p', MaxFieldLength + 1), new string('v', MaxFieldLength + 1) } },
+                Context = { InstrumentationKey = instrumentationKey }
+            };
+
+            // process in the opposite order to allow for an easier validation order
+            telemetryProcessor.Process(dependencyLong);
+            telemetryProcessor.Process(dependencyShort);
+
+            // ASSERT
+            Assert.AreEqual(
+                ((DependencyTelemetryDocument)accumulatorManager.CurrentDataAccumulator.TelemetryDocuments.ToList()[0]).Properties.First().Key,
+                dependencyShort.Properties.First().Key);
+            Assert.AreEqual(
+                ((DependencyTelemetryDocument)accumulatorManager.CurrentDataAccumulator.TelemetryDocuments.ToList()[0]).Properties.First().Value,
+                dependencyShort.Properties.First().Value);
+            Assert.AreEqual(
+                ((DependencyTelemetryDocument)accumulatorManager.CurrentDataAccumulator.TelemetryDocuments.ToList()[1]).Properties.First().Key,
+                dependencyShort.Properties.First().Key);
+            Assert.AreEqual(
+                ((DependencyTelemetryDocument)accumulatorManager.CurrentDataAccumulator.TelemetryDocuments.ToList()[1]).Properties.First().Value,
+                dependencyShort.Properties.First().Value);
```

Comment:
Cache the stuff that is expensive like .ToList() and .First() and reuse instead of recreating and reiterating.